dmoj/cptbox/compiler_isolate.py

```diff
@@ -14,7 +15,7 @@
 
 
 class CompilerIsolateTracer(IsolateTracer):
-    def __init__(self, *, tmpdir, read_fs, write_fs):
+    def __init__(self, *, tmpdir, read_fs: List[FilesystemAccessRule], write_fs: List[FilesystemAccessRule]):
```

Comment:
I know this is technically out of scope of the PR, but can you also add the type annotation for `tmpdir`? (IIRC it should be `str`).